### PR TITLE
Modernize Python versions

### PR DESCRIPTION
**Pull request by jugmac00**
- drop Python 3.4
- add Python 3.7 and Python 3.8
- fix pypy version for TravisCI

modified:   .travis.yml
modified:   CHANGES.rst
modified:   setup.py
modified:   tox.ini

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
+
 python:
   - 2.7
   - 3.4
   - 3.5
   - 3.6
-  - pypy-5.4.1
+  - 3.7
+  - 3.8-dev
+  - pypy3
```
Comment (mgedmin) [inline .travis.yml]: Why no `pypy`, just `pypy3`?

File: CHANGES.rst [hunk 2]
```diff
@@ -4,7 +4,11 @@ Change History
 4.0.1 (unreleased)
 ------------------
 
-- Nothing changed yet.
+- Add support for Python 3.7 and 3.8-dev.
+
+- Drop support for Python 3.4.
+
+- Fix PyPy version for TravisCI.
```
Comment (mgedmin) [inline CHANGES.rst]: Travis CI has both `pypy` and `pypy3` available for xenial.

File: .travis.yml (hunk 1 above)
Comment (jugmac00) [reply .travis.yml]: I have zero experience (and interest in pypy), so I just googled what pypy-5.4.1 gets replaced with and it seemed to me it is pypy3.

Added pypy, too.

File: CHANGES.rst (hunk 2 above)
Comment (jugmac00) [reply CHANGES.rst]: Now, both gets tested.

**Comment by mgedmin** [comment]
> Actually, I think it was me :-)

Very possible!  My memory is like a ... like a ... like a thing that forgets things.  Um.

**Pull request merged**